### PR TITLE
Remove some handlers from miners

### LINKED ISSUE #1916: Remove sharder endpoints from miners and miner endpoints from sharders
Due to a shared codebase:
- every miner endpoint listed on api.zus.network can be called on a sharder
- every sharder endpoint listed on api.zus.network can be called on a miner

Many times, these calls do not work as they are not intended to be called on a miner/sharder 
Sometimes they do work, but are flakey and are not guaranteed to work 

This causes confusion, and could lead apps to be built against the incorrect service, as we have seen with our own apps.

Acceptance criteria:
- Compile a list of miner and sharder concerns 
- Ensure only endpoints required for those concerns are accessible on the miner/sharder sservice 
- Chat to @stewartie4 to update api.zus.network
- Ensure our gosdk, wasm and all apps still work after this separation of concerns

### PR DESCRIPTION
**Pull request by Sriep**
## Fixes
This removes some of the handlers from miners:
* /v1/client/get/balance
* /v1/screst/
* /_nh/list/s

Removed from sharders:
* /v1/transaction/put
* /_nh/list/m

Put under an `is development` check for both miners and sharders
* /v1/scstats/
* /v1/config/get

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by stewartie4 — COMMENTED** [review]
This may also need updated https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/chaincore/chain/handler.go#L109

**Comment by peterlimg** [comment]
@Sriep we may keep handlers the following handlers on miners, they could be used to debug state and check config when needed:
- /v1/scstate/get
- /v1/config/get

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | trim_miner_handlers  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Review by stewartie4 — COMMENTED** [review]
Happy approve pending green test run (this is what caused me issues on my attempt at this issue!)

**Comment by dabasov** [comment]
> @Sriep we may keep handlers the following handlers on miners, they could be used to debug state and check config when needed:
> 
> * /v1/scstate/get
> * /v1/config/get

what do you think of putting them under IsDevelopment check? @peterlimg

File: code/go/0chain.net/chaincore/chain/state_handler.go [hunk 1]
```diff
@@ -60,15 +60,20 @@ func SetupScRestApiHandlers() {
 }
 
 /*SetupStateHandlers - setup handlers to manage state */
-func SetupStateHandlers() {
+func SetupSharderStateHandlers() {
 	c := GetServerChain()
 	http.HandleFunc("/v1/client/get/balance", common.WithCORS(common.UserRateLimit(common.ToJSONResponse(c.GetBalanceHandler))))
-	http.HandleFunc("/v1/scstate/get", common.WithCORS(common.UserRateLimit(common.ToJSONResponse(c.GetNodeFromSCState))))
```
Comment (peterlimg) [inline code/go/0chain.net/chaincore/chain/state_handler.go:66]: This endpoint is required for both miners and sharders, but as @dabasov suggested, we can put it under `development` endpoints, so as the `config/get`.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/chaincore/chain/handler.go [hunk 2]
```diff
@@ -57,10 +57,47 @@ func chainhandlersMap(c Chainer) map[string]func(http.ResponseWriter, *http.Requ
...
+				GetBlockHandler,
+			),
+		)
+	}
 
+	return m
+}
+
+func sharderHandlersMap(c Chainer) map[string]func(http.ResponseWriter, *http.Request) {
+	m := map[string]func(http.ResponseWriter, *http.Request){
```
Comment (peterlimg) [inline code/go/0chain.net/chaincore/chain/handler.go:77]: Sorry, I forget to mention that the endpoints of `/estimate_txn_fee` and `fees_table` endpoints are all miners endpoints, so as some other endpoints, such as `/get/latest_finalized` and `/latest_finalized_magic_block_summary`, etc. Basically all endpoints read data from MPT or data struct other than event db should be supported by both miners and sharders. We'd better keep them in common place of handlersMap and only move specific handlers to minerHandlerMap and sharderHandlersMap.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/chaincore/chain/handler_integration_tests.go [hunk 3]
```diff
@@ -51,15 +51,18 @@ func LatestFinalizedMagicBlockHandler(c Chainer) common.JSONResponderF {
 
 /*SetupHandlers sets up the necessary API end points */
 func SetupHandlers(c Chainer) {
-	hMap := handlersMap(c)
+	var hMap map[string]func(http.ResponseWriter, *http.Request)
 
 	if node.Self.Underlying().Type == node.NodeTypeMiner {
+		hMap := minerHandlersMap(c)
```
Comment (peterlimg) [inline code/go/0chain.net/chaincore/chain/handler_integration_tests.go]: Should not create new `hMap` here, the hMap will be empty. So let's use `hMap = minerHandlersMap(c)` instead

File: code/go/0chain.net/chaincore/chain/handler_integration_tests.go [hunk 4]
```diff
@@ -51,15 +51,18 @@ func LatestFinalizedMagicBlockHandler(c Chainer) common.JSONResponderF {
...
 		hMap[GetBlockV1Pattern] = BlockStats(
 			hMap[GetBlockV1Pattern],
 			BlockStatsConfigurator{
 				HashKey: "block",
 			},
 		)
+	} else {
+		hMap := handlersMap(c)
```
Comment (peterlimg) [inline code/go/0chain.net/chaincore/chain/handler_integration_tests.go]: Same as above

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | trim_miner_handlers  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | trim_miner_handlers  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Pull request merged**